jx/scripts/update-clients.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+ORG_REPOS=("jenkins-x/jx-ts-client")
+for org_repo in "${ORG_REPOS[@]}"; do
+  OUTDIR="$(jx step git fork-and-clone -b --print-out-dir https://github.com/$org_repo)"
+  echo "Forked repo to $OUTDIR"
+  VERSION="$(cat VERSION)"
+  pushd $OUTDIR
+  make all
+  git add -N .
+  git diff --exit-code
+  if [ $? -ne 0 ]; then
+    jx create pullrequest -b --push=true --body="upgrade $org_repo client to jx $VERSION" --title="upgrade to jx $VERSION" --label="updatebot"
+  fi
+  popd
+done
```

Comment:
So what is the difference really between the _hack_ directory and _jx/scripts_? Should not all scripts go under hack?

newline missing btw.

Reply:
And where and how is this script used then? It basically run make in a jenkins-x/jx-ts-client clone and create a pull request for any potential changes, right? OOI, where is this jx-ts-client used? For the UI?

Reply:
`hack` used to be how we did the client generation. I deleted it. I'm not sure why it got reintroduced.

Reply:
> And where and how is this script used then? This script will be used by the `jx` release build (eventually in a parallel step) to build and push changes to generated clients. Right now the only generated client we have is the one for typescript.

> It basically run make in a jenkins-x/jx-ts-client clone and create a pull request for any potential changes, right?

It only creates a pull request if changes are required but otherwise, yes.

> OOI, where is this jx-ts-client used? For the UI?

We will use it for that soon (once I have the build working)